torchtune/utils/multi_dataset.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Tuple
+
+from torch.utils.data import Dataset
+
+from torchtune import utils
+
+log = utils.get_logger("DEBUG")
+
+
+class MultiDataset(Dataset):
+    def __init__(self, datasets: List[List[Dataset]]):
+        self._datasets = datasets
+
+    def __getitem__(self, index: int) -> Tuple[List[int], List[int]]:
+        if index < 0 or index >= self.__len__():
```

Comment:
nit: reads a bit better
```suggestion
        if index < 0 or index >= len(self):
```